association.go

```diff
@@ -183,11 +185,11 @@ func newAssociation(rv reflect.Value, index int) Association {
 	}
 }
 
-func extractAssociationData(rt reflect.Type, index int) associationData {
+func extractAssociationData(rt reflect.Type, index []int) associationData {
 	var (
 		key = associationKey{
 			rt:    rt,
-			index: index,
+			index: fmt.Sprint(index),
```

Comment:
This could be replaced with a custom []int -> string function for better performance. Given, that those index slices are pretty short and contain small values, we can even hash them with a custom hashing function.

Reply:
how about printing the slice manually using `strconv` package
my concern is, `fmt.Sprint` is using interface that might be unnecessary here